### PR TITLE
[API] Add versioning compatibility to server and client

### PR DESCRIPTION
**Pull request by Tankilevitch**

### CODE REVIEW

File: mlrun/db/httpdb.py [hunk 1]
```diff
@@ -143,10 +162,12 @@ def api_call(
             :param json: JSON payload to be passed in the call
             :param headers: REST headers, passed as a dictionary: ``{"<header-name>": "<header-value>"}``
             :param timeout: API call timeout
+            :param version: API version
```
Comment (Hedingber) [inline mlrun/db/httpdb.py:165]:
```suggestion
            :param version: API version to use, None (the default) will mean to use the default value from mlconf
```

File: tests/api/api/feature_store/base.py [hunk 2]
```diff
@@ -137,11 +137,11 @@ def _test_partition_by_for_feature_store_objects(
...
     object_url_name = object_name.replace("_", "-")
     response = client.get(
-        f"/api/projects/{project_name}/{object_url_name}?partition-by=name"
+        f"projects/{project_name}/{object_url_name}?partition-by=name"
     )
     assert response.status_code == HTTPStatus.BAD_REQUEST.value
     # An invalid partition-by field - will be failed by fastapi due to schema validation.
     response = client.get(
-        f"/api/projects/{project_name}/{object_url_name}?partition-by=key&partition-sort-by=name"
+        f"projects/{project_name}/{object_url_name}?partition-by=key&partition-sort-by=updated"
```
Comment (Hedingber) [inline tests/api/api/feature_store/base.py:145]: Why you're changing the sort-by ?

**Review by Hedingber — CHANGES_REQUESTED** [review]
Very nice, few minor comments

File: mlrun/api/main.py [hunk 3]
```diff
@@ -32,19 +32,27 @@
 from mlrun.runtimes import RuntimeKinds, get_runtime_handler
 from mlrun.utils import logger
 
+API_PREFIX = "/api"
+BASE_VERSIONED_API_PREFIX = (
```
[the remark this answers is not in the document]
Comment (Tankilevitch) [reply mlrun/api/main.py:36]: Maybe the condition is redundant, but I do think `BASE_VERSIONED_API_PREFIX` is required.
I wanted one location where it was configured and all components that needed to know the prefix, such as the paths to the docs and the TestClient, could point to.

File: mlrun/db/httpdb.py [hunk 4]
```diff
@@ -119,6 +119,23 @@ def __repr__(self):
...
 
+    @staticmethod
+    def get_api_prefix_url(version: str = None, versioned: bool = True) -> str:
+        if not versioned:
+            return "api"
+        version = version if version else mlrun.mlconf.api_base_version
+        # The next line will be removed right when we stop support un-versioned api
+        # TODO: remove line when un-versioned api is deprecated
+        api_version_path = f"api/{version}" if version else "api"
+        return api_version_path
```
[the remark this answers is not in the document]
Comment (Tankilevitch) [reply mlrun/db/httpdb.py:130]: I added the `versioned` param to let the function know whether he really wants the unversioned path or if it doesn't matter to him and it wants to get the default from the `api_base_version`. If I understand correctly what you recommend is that the user will define the exact version he wants regardless of the default one?

**Review by Hedingber — APPROVED** [review]

**Pull request merged**